apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -129,16 +129,23 @@ handle_request('GetAccountBalance', Req, _Context) ->
               {ok, PK} = aec_keys:pubkey(),
               PK;
           PK ->
-              base64:decode(PK)
+              try base64:decode(PK)
+              catch error:{badarg, _} -> not_base64_encoded
```

Comment:
Ok.

----

I tested locally:
```
4> f(), try base64:decode(<<"foo">>) catch E:R -> {E, R} end.
{error,{badmatch,false}}
```

Reply:
(Just to be clear: no action on this comment - this comment states that this line looks correct.)

Reply:
I correct myself: there is issue. base64:decode returns two kinds of errors: badmatch and badarg.

Reply:
```
6> f(), try base64:decode(<<"bob_address">>) catch E:R -> {E,R} end.
{error,{badarg,95}}
7> f(), try base64:decode(<<"bobaddress">>) catch E:R -> {E,R} end.
{error,{badmatch,false}}
```

Reply:
So in one case 400 is returned, in other case 500 is returned.
```
$ curl -v http://localhost:3013/v1/account/balance?pub_key=bobaddress
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 3013 (#0)
> GET /v1/account/balance?pub_key=bobaddress HTTP/1.1
> Host: localhost:3013
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 500 Internal Server Error
< server: Cowboy
< date: Wed, 29 Nov 2017 12:09:18 GMT
< content-length: 0
< content-type: application/json
<
* Connection #0 to host localhost left intact

$ curl -v http://localhost:3013/v1/account/balance?pub_key=bob_address
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 3013 (#0)
> GET /v1/account/balance?pub_key=bob_address HTTP/1.1
> Host: localhost:3013
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 400 Bad Request
< server: Cowboy
< date: Wed, 29 Nov 2017 12:09:21 GMT
< content-length: 28
< content-type: application/json
<
* Connection #0 to host localhost left intact
```

Reply:
So this PR catches the case when the pubkey specified contains chars not in the base64 range. Though does not catch the case that the chars are in the base64 range though it cannot be decoded.